assistant.go

```diff
@@ -0,0 +1,160 @@
+// @see https://api.slack.com/methods/assistant.threads.setSuggestedPrompts
+func (api *Client) SetAssistantThreadsSuggestedPrompts(params AssistantThreadsSetSuggestedPromptsParameters) (err error) {
+	return api.SetAssistantThreadsSuggestedPromptsContext(context.Background(), params)
+}
+
+// SetAssistantThreadSuggestedPromptsContext sets the suggested prompts for a thread with a custom context
+// @see https://api.slack.com/methods/assistant.threads.setSuggestedPrompts
+func (api *Client) SetAssistantThreadsSuggestedPromptsContext(ctx context.Context, params AssistantThreadsSetSuggestedPromptsParameters) (err error) {
+
+	values := url.Values{
+		"token": {api.token},
+	}
+
+	if params.ChannelID != "" {
+		values.Add("channel_id", params.ChannelID)
+	}
+
+	if params.ThreadTS != "" {
+		values.Add("thread_ts", params.ThreadTS)
+	}
+
+	// Send Prompts as JSON
+	prompts, err := json.Marshal(params.Prompts)
+	if err != nil {
+		return err
+	}
+
+	values.Add("prompts", string(prompts))
+
+	response := struct {
+		SlackResponse
+	}{}
+
+	err = api.postMethod(ctx, "assistant.threads.setSuggestedPrompts", values, &response)
+	if err != nil {
+		return
+	}
+
+	return response.Err()
+}
+
+// SetAssistantThreadStatus sets the status of a thread
+// @see https://api.slack.com/methods/assistant.threads.setStatus
+func (api *Client) SetAssistantThreadsStatus(params AssistantThreadsSetStatusParameters) (err error) {
+	return api.SetAssistantThreadsStatusContext(context.Background(), params)
+}
+
+// SetAssistantThreadStatusContext sets the status of a thread with a custom context
+// @see https://api.slack.com/methods/assistant.threads.setStatus
+func (api *Client) SetAssistantThreadsStatusContext(ctx context.Context, params AssistantThreadsSetStatusParameters) (err error) {
+
+	values := url.Values{
+		"token": {api.token},
+	}
+
+	if params.ChannelID != "" {
+		values.Add("channel_id", params.ChannelID)
+	}
+
+	if params.ThreadTS != "" {
+		values.Add("thread_ts", params.ThreadTS)
+	}
+
+	// Always send the status parameter, if empty, it will clear any existing status
+	values.Add("status", params.Status)
+
+	response := struct {
+		SlackResponse
+	}{}
+
+	err = api.postMethod(ctx, "assistant.threads.setStatus", values, &response)
+	if err != nil {
+		return
+	}
+
+	return response.Err()
+}
+
+// SetAssistantThreadsTitle sets the title of a thread
+// @see https://api.slack.com/methods/assistant.threads.setTitle
+func (api *Client) SetAssistantThreadsTitle(params AssistantThreadsSetTitleParameters) (err error) {
+	return api.SetAssistantThreadsTitleContext(context.Background(), params)
+}
+
+// SetAssistantThreadsTitleContext sets the title of a thread with a custom context
+// @see https://api.slack.com/methods/assistant.threads.setTitle
+func (api *Client) SetAssistantThreadsTitleContext(ctx context.Context, params AssistantThreadsSetTitleParameters) (err error) {
+
+	values := url.Values{
+		"token": {api.token},
+	}
+
+	if params.ChannelID != "" {
```

Comment:
I don't think this would ever happen since the fields are required both in the API call and the input struct - can you confirm?